pandas/core/strings/accessor.py

```diff
@@ -2145,7 +2157,7 @@ def startswith(self, pat, na=None):
         dtype: bool
         """
         result = self._data.array._str_startswith(pat, na=na)
-        return self._wrap_result(result, returns_string=False)
+        return self._wrap_result(result, returns_string=False, returns_bool=True)
```

Comment:
can we just change _wrap_result so we don't have 2 keywords? maybe rename to returns_dtype and pass a string?

Reply:
thats tough bc the end-result dtype depends on self._is_string

Reply:
Looking at this again it really doesnt look worth the trouble to try to combine these args

Reply:
I don't think either keywords should really be necessary here, the issue is that the arrays passed when expand is True are object arrays of lists/tuples

which method returns bool, where the dtype of the result is not bool?

Reply:
> which method returns bool, where the dtype of the result is not bool?

makes sense, ill give this a try

Reply:
e.g `isnumeric` we can get object-dtype result of bools and nans

Reply:
the wrapped result would be object in this case? This is for a object dtype array? for StringArray/ArrayString array the return type should always be boolean. object dtype accessors do cast their results, except extract, so if the result is all bools and no nans, the result dtype should be np.bool

Reply:
> the result dtype should be np.bool

the result to be wrapped dtype should be np.bool

Reply:
these conversions are handled in _str_map in ObjectStringArrayMixin